### ISSUE TITLE
Generalizing Hand Segmentation in Egocentric Videos with Uncertainty-Guided Model Adaptation, CVPR2020

### ISSUE THREAD

**Issue by lisosia** [lisosia/cv_knowledge#13]
## 一言でいうと
別データセットへのDomain Adaptionのために、Pseudo-Labelling を使った、複数IterationのTrainingをしたい。
Bayesian CNN の出力するUncertaintyを（ピクセル間）lossの重みとする（k-1 round目でのUncertaintyが、k round目のtrain時に使用される）。
Bayesian CNN の出力するUncertaintyはここでは、nn.Dropout を有効にして複数回forwardした時の分散。
ちなみに普通のCNNでsoftmax結果をUncertaintyとしても効果あり。

![image](https://user-images.githubusercontent.com/8558911/91456293-0e536c80-e8be-11ea-8715-743e253dd1f7.png)
![image](https://user-images.githubusercontent.com/8558911/91456375-24f9c380-e8be-11ea-918d-2bd62e6c1573.png)
![image](https://user-images.githubusercontent.com/8558911/91456424-393dc080-e8be-11ea-8fdf-d9548b190a99.png)

k round目のLoss. ここで、Uncertaintyは[0,1]でnormalizeされる
![image](https://user-images.githubusercontent.com/8558911/91456582-6be7b900-e8be-11ea-9922-1ab6fbad16fa.png)
![image](https://user-images.githubusercontent.com/8558911/91456635-7904a800-e8be-11ea-9c96-891fdc39f78c.png)

Procedure
![image](https://user-images.githubusercontent.com/8558911/91456797-a7828300-e8be-11ea-9d65-9891a77e4532.png)

### 論文リンク
https://openaccess.thecvf.com/content_CVPR_2020/papers/Cai_Generalizing_Hand_Segmentation_in_Egocentric_Videos_With_Uncertainty-Guided_Model_Adaptation_CVPR_2020_paper.pdf

### 著者/所属機関
Hunan University, etc

### 投稿日付(yyyy/MM/dd)
2020/

## 結果
1行目 vs 2行目, 3行目vs4行目⇢Pseudo Labelling の効果
2行目 vs 4行目⇢Uncertainty Guide として Baysesian CNN を用いる有用性

![image](https://user-images.githubusercontent.com/8558911/91456886-be28da00-e8be-11ea-89e5-8c611d0826a2.png)
![image](https://user-images.githubusercontent.com/8558911/91456957-d1d44080-e8be-11ea-85ff-d48aea87d130.png)

**Comment by lisosia** [comment]
実装：https://github.com/cai-mj/UMA

Bottleneck に torch.nn.Dropout(p=0.5) が使われている
mask updateのときの bayesian CNN のforward回数は10 https://github.com/cai-mj/UMA/blob/master/train_bayesian_adapt.py#L190
